apps/core/lib/core/event_repo.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Core.EventRepo do
+  @type operation ::
+          :user_created
+          | :user_updated
+          | :user_deleted
+          | :post_created
+
+@spec save_event(operation, {pos_integer(), String.t(), any()}) :: :ok | {:error, any()}
+  def save_event(:user_created, {user_id, user_email, time}),
+    do: write_to_mnesia(user_id, :user_created, %{user_email: user_email}, time)
+
+  def save_event(:user_updated, {user_id, attrs, time}),
+    do: write_to_mnesia(user_id, :user_updated, attrs, time)
+
+  def save_event(:user_deleted, {user_id, time}),
+    do: write_to_mnesia(user_id, :user_deleted, %{}, time)
+
+  def save_event(:post_created, {user_id, post_id, post_body, time}),
+    do:
+      write_to_mnesia(
+        user_id,
+        :post_created,
+        %{user_id: user_id, post_id: post_id, post_body: post_body},
```

Comment:
Putting `user_id` into the event's content seems pointless since it's already included in its "key".

Reply:
@mkwiatkowski OO thinking got me here ;)
@smt116 I think that `write` and `read` are too vague names. But I agree they should not have `mnesia`